pkg/broker/binding_operations.go

```diff
@@ -139,20 +141,36 @@ func userFromParams(bindingID string, password string, rawParams []byte) (*atlas
 		}
 	}
 
-	// Set binding ID as username and add password.
-	params.User.Username = bindingID
-	params.User.Password = password
-
-	// If no role is specified we default to read/write on any database.
-	// This is the default role when creating a user through the Atlas UI.
-	if len(params.User.Roles) == 0 {
-		params.User.Roles = []atlas.Role{
-			atlas.Role{
-				Name:         "readWriteAnyDatabase",
-				DatabaseName: "admin",
-			},
+	// If the plan ID is specified we construct the provider object from the service and plan.
+	// The plan ID is required for binding
+	if planID != "" {
+		provider, err := b.findProviderByServiceID(serviceID)
+		if err != nil {
+			return nil, err
 		}
+
+		_, err = findInstanceSizeByPlanID(provider, planID)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set binding ID as username and add password.
+		params.User.Username = bindingID
+		params.User.Password = password
+
+		// If no role is specified we default to read/write on any database.
+		// This is the default role when creating a user through the Atlas UI.
+		if len(params.User.Roles) == 0 {
+			params.User.Roles = []atlas.Role{
+				atlas.Role{
+					Name:         "readWriteAnyDatabase",
+					DatabaseName: "admin",
+				},
+			}
+		}
+
+		return params.User, nil
 	}
 
-	return params.User, nil
+	return nil, apiresponses.NewFailureResponse(errors.New("Invalid plan ID"), http.StatusBadRequest, "invalid-plan-id")
```

Comment:
Removes this as well when removing the if statement.

Reply:
Got it!